pyproject.toml

```diff
@@ -1,7 +1,97 @@
 [build-system]
-requires = ["jupyter_packaging~=0.7.9", "jupyterlab>=3,<=4", "setuptools>=40.8.0", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling>=1.5.0", "hatch-nodejs-version"]
+build-backend = "hatchling.build"
+
+[project]
+name = "jupytext"
+description = "Jupyter notebooks as Markdown documents, Julia, Python or R scripts"
+license = { file = "LICENSE" }
+authors = [
+    { name = "Marc Wouts", email = "marc.wouts@gmail.com" },
+]
+readme = "README.md"
+requires-python = ">=3.8"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "License :: OSI Approved :: MIT License",
+    "Environment :: Console",
+    "Framework :: Jupyter",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Topic :: Text Processing :: Markup",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dependencies = [
+    "nbformat",
+    "pyyaml",
+    "toml",
+    "markdown-it-py>=1.0.0",
+    "mdit_py_plugins",
+]
+dynamic = ["version"]
+
+[project.urls]
+homepage = "https://github.com/mwouts/jupytext"
+source = "https://github.com/mwouts/jupytext"
+documentation = "https://jupytext.readthedocs.io/en/latest/index.html"
+
+[project.optional-dependencies]
+extension = [
+    "jupytext_extension",
+]
+dev = [
+    "autopep8",
+    "black",
+    "isort",
+    "flake8",
+    "pytest",
+    "pytest-cov>=2.6.1",
+    "gitpython",
+    "jupyterlab",
+    "notebook",
+    "nbconvert",
+    "ipykernel",
+]
+# left for back-compatibility
+rst2md = [
+    "sphinx-gallery~=0.7.0",
+]
+
+[project.scripts]
+"jupytext" = "jupytext.cli:jupytext"
+"jupytext-config" = "jupytext.jupytext_config:main"
+
+[tool.hatch.version]
+path = "jupytext/version.py"
+
+[tool.hatch.build.targets.wheel.shared-data]
+"jupyter-config" = "etc/jupyter"
+"jupytext/nbextension" = "share/jupyter/nbextensions/jupytext"
```

Comment:
@mwouts How are these files supposed to be handled? Aren't these specific to the jupyterlab extension?

My understanding is that the top-level main project should be a pure python project and all jupyter specific stuff should be mygrated to `jupiterlab-jupytext`

Reply:
The files in the `jupyter-config` are config files for server extension that currently core `jupytext` provides. I agree with you that we can move this server extension to `jupyterlab_jupytext` package as well. In that case, maybe we need to rename the package to `jupyter_jupytext` as server extension is meant for `jupyter_server` and not just `jupyterlab`?

I agree that we should move `nbextension` to `jupyterlab_jupytext` package. However, this `nbextension` works only for Notebook<7 and as we are moving to JupyterLab 4 and Notebook 7, I do not know if it still makes sense to distribute these extension files. 

@mwouts What do you think?

Reply:
Yes, many good ideas here!

Sure `jupytext-config` would fit well within the Jupyter extension.

Re the name I do prefer `jupyterlab-jupytext` over `jupyter-jupytext` because it's consistent with the name that we have used until now.

Also, yes we can remove the `nbextension` files now, no need to distribute them anymore.